vendor/views/vendor.py

```diff
@@ -104,15 +104,11 @@ def session_cart(self, request, offer):
         return session_cart
 
     def post(self, request, *args, **kwargs):
-        site = get_current_site(request)
-        if hasattr(request, 'site'):
-            site = request.site
-
-        offer = Offer.objects.get(site=site, slug=self.kwargs["slug"])
+        offer = Offer.objects.get(site=request.site, slug=self.kwargs["slug"])
         if request.user.is_anonymous:
             request.session['session_cart'] = self.session_cart(request, offer)
         else:
-            profile, created = self.request.user.customer_profile.get_or_create(site=get_current_site(request))
+            profile, created = self.request.user.customer_profile.get_or_create(site=request.site)
```

Comment:
use the new function here to get the site object instead of modifying the request object.

Reply:
changed to: `site=get_site_from_request(request)`